cmd/waas-auth/main.go

```diff
@@ -20,26 +23,32 @@ func main() {
 		panic(err)
 	}
 
-	// HTTP client to use for all outgoing connections out of the enclave
-	client := http.DefaultClient
+	baseTransport := http.DefaultTransport
 	if cfg.Service.VSock {
-		client = &http.Client{
-			Transport: &http.Transport{
-				Proxy: func(_ *http.Request) (*url.URL, error) {
-					return url.Parse("http://vsock-proxy")
-				},
-				DialContext: func(ctx context.Context, network string, addr string) (net.Conn, error) {
-					log.Printf("Outgoing connection to %s://%s\n", network, addr)
-					return vsock.Dial(3, cfg.Service.ProxyPort, nil)
-				},
-				MaxIdleConns:          100,
-				IdleConnTimeout:       90 * time.Second,
-				TLSHandshakeTimeout:   10 * time.Second,
-				ExpectContinueTimeout: 1 * time.Second,
+		baseTransport = &http.Transport{
+			Proxy: func(_ *http.Request) (*url.URL, error) {
+				return url.Parse("http://vsock-proxy")
+			},
+			DialContext: func(ctx context.Context, network string, addr string) (net.Conn, error) {
+				log.Printf("Outgoing connection to %s://%s\n", network, addr)
+				return vsock.Dial(3, cfg.Service.ProxyPort, nil)
 			},
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
 		}
 	}
 
+	// HTTP client to use for all outgoing connections out of the enclave
+	client := &http.Client{
+		Transport: transport.Chain(
+			baseTransport,
+			transport.SetHeader("User-Agent", "waas-authenticator/"+waasauthenticator.VERSION),
```

Comment:
I think we should avoid overwriting the User-Agent, and if you want to pass this information we should put it in another header, such as like "Server-Agent" ... or something like that

The User-Agent will contain the end-user's agent, which will be useful for us and we don't want to override that

Reply:
I looked at what Cloudflare does, and they use the "Server" header to announce their server.

So I propose, just rename above "User-Agent" header to "Server"